core/src/main/java/com/google/zxing/oned/Code93Writer.java

```diff
@@ -125,4 +100,66 @@ private static int computeChecksumIndex(String contents, int maxWeight) {
     }
     return total % 47;
   }
+
+  private static String convertToExtended(String contents) {
+    int length = contents.length();
+    StringBuilder extendedContent = new StringBuilder(length * 2);
+    for (int i = 0; i < length; i++) {
+      char character = contents.charAt(i);
+      if (character == 0) {
+        // NUL: (%)U
+        extendedContent.append("bU");
+      } else if (character <= 26) {
+        // SOH - SUB: ($)A - ($)Z
+        extendedContent.append('a');
```

Comment:
I thought these were prefixed with $? or is https://en.wikipedia.org/wiki/Code_93#Full_ASCII_Code_93 not the right reference?

Reply:
Please see `Code93Reader.ALPHABET_STRING`.
Control chars are mapped to `abcd`.

I add a comment in this method.